osu.Game/Rulesets/Edit/Checks/CheckHitsoundsFormat.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using ManagedBass;
+using osu.Framework.Audio.Callbacks;
+using osu.Game.Beatmaps;
+using osu.Game.Extensions;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckHitsoundsFormat : ICheck
+    {
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Audio, "Checks for hitsound formats.");
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateFormatUnsupported(this),
+            new IssueTemplateIncorrectFormat(this),
+        };
+
+        private IEnumerable<ChannelType> allowedFormats => new ChannelType[]
+        {
+            ChannelType.WavePCM,
+            ChannelType.WaveFloat,
+            ChannelType.OGG,
+            ChannelType.Wave | ChannelType.OGG,
```

Comment:
How were these flags selected? Why are `WavePCM` and `WaveFloat` specifically here but not plain `Wave` which the docs state is the proper way to check for any wave file?

Reply:
We can take 3 beatmaps for example here (although does happen with most of them): 
https://osu.ppy.sh/beatmapsets/773455#osu/1625870
https://osu.ppy.sh/beatmapsets/1842041#osu/3783525
https://osu.ppy.sh/beatmapsets/1529362#osu/3219157

if we remove `WavePCM` and `WaveFloat` and replace it with `Wave` most beatmap hitsounds will fail this check, that's why i included them and not just plain `Wave`

![image](https://github.com/ppy/osu/assets/90941580/321d6dce-a0f0-4667-9fbb-73216be794cc)

Reply:
Let me try again then.

- `Wave` has a value of `0x40000`.
- [BASS docs](https://www.un4seen.com/doc/#bass/BASS_CHANNELINFO.html) say this:

    > WAVE format flag. This can be used to test if the channel is any kind of WAVE format. The codec (the file's "wFormatTag") is specified in the LOWORD. Additional information is also avaliable from [BASS_ChannelGetTags](https://www.un4seen.com/doc/bass/BASS_ChannelGetTags.html) (BASS_TAG_WAVEFORMAT).
- `BASS_CTYPE_STREAM_WAV_PCM` is specified there as a separate value, but it ends up being `0x50001`, which is `Wave | Stream | 0x1`.
- Similarly, `BASS_CTYPE_STREAM_WAV_FLOAT` is `0x50003`, which is `Wave | Stream | 0x3`.

So I guess my point here would be: why is this checking any specific types or even attempting to do `Wave | OGG` (what does that even mean?!), rather than doing `type & Wave > 0`?

Reply:
> So I guess my point here would be: why is this checking any specific types or even attempting to do Wave | OGG and OGG (what does that even mean?!), rather than doing type & Wave > 0 or something?

Thanks for clarifying, well that was ignorance of my part. To fix this i removed the `allowedFormats` field and changed the condition to the following:
```csharp
var audioInfo = Bass.ChannelGetInfo(decodeStream);

if ((audioInfo.ChannelType & (ChannelType.Wave | ChannelType.Recording)) == 0)
    yield return new IssueTemplateIncorrectFormat(this).Create(file.Filename);

Bass.StreamFree(decodeStream);
```
I used `Recording` to identify a OGG format because using `ChannelType.OGG` itself was identifying mp3 as a allowed format because of it containing `ChannelType.Stream` i think. Let me know if this looks fine.

Reply:
I think the better check would be something along the lines of

```csharp
audioInfo.ChannelType & ChannelType.Wave == 0 && audioInfo.ChannelType != ChannelType.OGG
```

Doing that recording thing feels pretty wrong if `0x2` is explicitly documented as meaning something else. `0x2` only has the "ogg" meaning in combination with other flags that have the low word cleared.